File: Java/Strategy/src/main/java/DecoyDuck.java
package Strategy;

public class DecoyDuck extends Duck {
	//Constructor
	public DecoyDuck() {
		quackBehaviour =  new Squeak();
		flyBehaviour = new FlyWithWings();
	}

	public void display() {
		System.out.println("I'm a real Mallard Duck");
	}

}

File: CSharp/Strategy/Ducks/MiniDuckSimualtor/DecoyDuck.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Strategy
{
    public class DecoyDuck : Duck
    {
        public DecoyDuck()
        {
            flyBehaviour = new FlyWithWings();
            quackBehaviour = new Quack();
        }

        public override void display()
        {
            Console.Out.WriteLine("I am a Decoy Duck");
        }
    }
}
